haystack/evaluation/eval_run_result.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import Any, Dict, List
+from warnings import warn
+
+from pandas import DataFrame
+from pandas import concat as pd_concat
+
+
+class BaseEvaluationRunResult(ABC):
```

Comment:
Isn't the `EvaluationRunResult` too specific to need an abstraction class?

Reply:
I see your point - also, I can imagine that we probably need to adjust a bit the methods (in the base class) that each extension needs to implement - anyway, don't think this is an issue. Let's merge it 👍🏽